pkg/ui/console/client.go

```diff
@@ -92,7 +93,7 @@ func (c Client) DownloadProviders(ctx context.Context) error {
 		c.updater.Wait()
 	}
 	ui.ColorizedOutput(ui.ColorProgress, "Finished provider initialization...\n\n")
-	updates, _ := c.c.CheckForProviderUpdates(ctx)
+	updates := client.CheckForProviderUpdates(ctx, registry.GetLatestRelease, c.c.Providers, c.c.Logger)
```

Comment:
why migrate it from method level and then pass everything as parameters?

Reply:
It makes it more of a pure function, easier to test and easier to reason about. Doesn't matter much in this case, I reverted it.